ecosystem/indexer-grpc/indexer-transaction-generator/src/transaction_generator.rs

```diff
@@ -76,12 +97,185 @@ impl TransactionGenerator {
         self.managed_node.stop().await
     }
 
+    /// Go through the test case steps and generate the transactions.
     async fn generate(&self, test_case: TestCase) -> anyhow::Result<()> {
-        // TODO: finish the test generation logic.
         println!(
-            "Generate transaction for test case {} at {:?} ",
+            "\n Generate transaction for test case {} at {:?} \n",
             test_case.name, self.output_folder,
         );
+        // Test case output folder.
+        let test_case_output_folder = self.output_folder.join(&test_case.name);
+        // If the output folder doesn't exist, create it.
+        if !test_case_output_folder.exists() {
+            std::fs::create_dir_all(&test_case_output_folder)?;
+        }
+        let mut transactions_version_to_capture = vec![];
+        // Execute the transactions.
+        for step in test_case.steps {
+            match step {
+                Step::Setup((path, _, config)) => {
+                    // Change current directory to the setup script folder.
+                    std::env::set_current_dir(&path)
+                        .context(format!("Failed to change directory to: {:?}", path))?;
+                    if let Some(test_config) = config {
+                        if let Some(amount) = test_config.fund_amount {
+                            println!("Funding account with amount: {}", amount);
+                            let fund = FundWithFaucet::new_for_indexer_testing(amount);
+                            fund.execute()
+                                .await
+                                .context("Failed to fund the account.")?;
+                            println!("Funded account with amount: {}", amount);
```

Comment:
Not a big deal but it might be better to log rather than print, that way you can get logging levels, see where the log line comes from, etc. I know for CLI tools printing is more standard though, so up to you.